pkg/cpi/mockingUtils.go

```diff
@@ -77,6 +79,24 @@ func GetEmptyHTTPResponseBodyAndErrorNil() (*http.Response, error) {
 	return &res, nil
 }
 
+//StoreFileInOsAndErrorNil - Method to store read artifacts in the file system
+func StoreFileInOsAndErrorNil(FilePath string, resp *http.Response) error {
```

Comment:
As far as I can see, only the positive case is tested here. Maybe you want to test this function directly, instead of via the APIProviderDownload step.

Reply:
Idea is to write a common function to store any file in OS which can also be consumed by other steps in future. So, testing with the corresponding step would makes sense in my opinion.